### PR TITLE
Bump torch to <2.4.1

### PR DESCRIPTION
**Pull request by eitanturok**
**Update:** Running the latest tests manually and they pass [here](https://github.com/databricks/megablocks/actions/runs/10633150010)

# What does this PR do?
* use "torch >= 2.4.0, < 2.4.1" in setup.py, pyproject.toml
* use forks of grouped_gemm, stk that support torch2.4
	* will update to official versions of groupped_gemm, stk once they are released
* use newer version of composer that supports torch2.4
	* will use official version of composer once it is released
* tests use torch2.4 image

This Github Actions here run the old tests and predictably fail. I manually ran the new Github Actions [here](https://github.com/databricks/megablocks/actions/runs/10564911758) and they all pass.

# What issue(s) does this change relate to?

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md)?
- [x] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Comment by eitanturok** [comment]
blocked by waiting for new releases of composer, groupped_gemm, stk

File: setup.py [hunk 1]
```diff
@@ -83,7 +83,7 @@
 ]
 
 extra_deps['testing'] = [
-    'mosaicml>=0.22.0',
+    'mosaicml>=0.23.6',
```
Comment (snarayan21) [inline setup.py:86]:
```suggestion
    'mosaicml>=0.24.1',
```
should be 0.24.1

**Review by snarayan21 — COMMENTED** [review]
compsoer version

**Review by snarayan21 — APPROVED** [review]
lgtm

**Comment by eitanturok** [comment]
Manually ran the tests again and they are passing/failing [here](https://github.com/databricks/megablocks/actions/runs/10622364966)

**Review by mihir-db — APPROVED** [review]

**Pull request merged**